src/engine/guidance/collapse_turns.cpp

```diff
@@ -284,6 +284,97 @@ void StaggeredTurnStrategy::operator()(RouteStep &step_at_turn_location,
                                                                      : TurnType::NewName;
 }
 
+void CombineSegregatedStepsStrategy::operator()(RouteStep &step_at_turn_location,
+                                                const RouteStep &transfer_from_step) const
+{
+    // TODO
+    if (hasTurnType(step_at_turn_location, TurnType::EndOfRoad) ||
+        hasTurnType(transfer_from_step, TurnType::EndOfRoad))
+    {
+        setInstructionType(step_at_turn_location, TurnType::EndOfRoad);
+    }
+}
+
+SegregatedTurnStrategy::SegregatedTurnStrategy(const RouteStep &step_prior_to_intersection)
+    : step_prior_to_intersection(step_prior_to_intersection)
+{
+}
+
+void SegregatedTurnStrategy::operator()(RouteStep &step_at_turn_location,
+                                        const RouteStep &transfer_from_step) const
+{
+    const auto calculate_turn_direction = [](const RouteStep &entry_step,
+                                             const RouteStep &exit_step) {
+        const double angle = util::bearing::angleBetween(entry_step.maneuver.bearing_before,
+                                                         exit_step.maneuver.bearing_after);
+
+        return getTurnDirection(angle);
+    };
+
+    // Calculate turn direction for segregated
+    const auto turn_direction = calculate_turn_direction(step_at_turn_location, transfer_from_step);
+
+    const auto is_straight_step = [](const RouteStep &step) {
+        return ((hasTurnType(step, TurnType::NewName) || hasTurnType(step, TurnType::Continue) ||
+                 hasTurnType(step, TurnType::Suppressed) || hasTurnType(step, TurnType::Turn)) &&
+                (hasModifier(step, DirectionModifier::Straight) ||
+                 hasModifier(step, DirectionModifier::SlightLeft) ||
+                 hasModifier(step, DirectionModifier::SlightRight)));
+    };
+
+    const auto is_turn_step = [](const RouteStep &step) {
+        return (hasTurnType(step, TurnType::Turn) || hasTurnType(step, TurnType::Continue) ||
+                hasTurnType(step, TurnType::NewName) || hasTurnType(step, TurnType::Suppressed));
+    };
+
+    // Process end of road step
+    if (hasTurnType(step_at_turn_location, TurnType::EndOfRoad) ||
+        hasTurnType(transfer_from_step, TurnType::EndOfRoad))
+    {
+        // Keep end of road
+        setInstructionType(step_at_turn_location, TurnType::EndOfRoad);
+
+        // Set modifier based on turn angle
+        setModifier(step_at_turn_location, turn_direction);
+    }
+    // TODO maybe remove fork logic after updating identification logic
+    // Process fork step at turn
+    else if (hasTurnType(step_at_turn_location, TurnType::Fork))
+    {
+        // Keep fork
+        setInstructionType(step_at_turn_location, TurnType::Fork);
```

Comment:
`step_at_turn_location` has already `TurnType::Fork`
What will happen if  `hasTurnType(transfer_from_step, TurnType::Fork)` is true?